agent/src/beerocks/slave/link_metrics/ieee802_3_link_metrics_collector.cpp

```diff
@@ -0,0 +1,279 @@
+                             sLinkMetrics &link_metrics)
+{
+    bool result = false;
+
+    struct ifinfomsg *iface = static_cast<ifinfomsg *>(NLMSG_DATA(h));
+
+    size_t length = 0;
+    if (h->nlmsg_len > NLMSG_LENGTH(sizeof(*iface))) {
+        length = h->nlmsg_len - NLMSG_LENGTH(sizeof(*iface));
+    }
+
+    /**
+     * Loop over all attributes of the RTM_NEWLINK message
+     */
+    for (struct rtattr *attribute = IFLA_RTA(iface); RTA_OK(attribute, length);
+         attribute                = RTA_NEXT(attribute, length)) {
+        switch (attribute->rta_type) {
+        case IFLA_IFNAME:
+            /**
+             * This message contains the stats for the interface we are interested in
+             */
+            if (0 == std::strncmp(local_interface_name.c_str(), (char *)RTA_DATA(attribute),
+                                  local_interface_name.length() + 1)) {
+                result = true;
+            }
+            break;
+        case IFLA_STATS:
+            if (result) {
+                struct rtnl_link_stats *stats = (struct rtnl_link_stats *)RTA_DATA(attribute);
+
+                /**
+                 * Get interface speed into PHY rate.
+                 */
+                uint32_t phy_rate_mbps = UINT32_MAX;
+                beerocks::net::network_utils::linux_iface_get_speed(local_interface_name,
+                                                                    phy_rate_mbps);
+
+                link_metrics.transmitter.packet_errors       = stats->tx_errors;
+                link_metrics.transmitter.transmitted_packets = stats->tx_packets;
+                /**
+                 * Note: The MAC throughput capacity is a function of the physical data rate and
+                 * of the MAC overhead. We could somehow compute such overhead or, for simplicity,
+                 * set the MAC throughput as a percentage of the physical data rate.
+                 * For Ethernet, we can estimate the overhead: 7 bytes preamble, 1 byte SFD, 14
+                 * bytes header, 4 bytes CRC and 12 bytes of interpacket gap on a 1500 byte
+                 * payload. So 1500/1538.
+                 * (see https://en.wikipedia.org/wiki/Ethernet_frame)
+                 */
+                const uint32_t layer2_payload_size = 1500;
+                const uint32_t layer1_total_size   = 1538;
+                link_metrics.transmitter.mac_throughput_capacity_mbps =
+                    phy_rate_mbps * ((float)layer2_payload_size / (float)layer1_total_size);
+                // Note: For simplicity, link availability is set to "100% of the time"
+                link_metrics.transmitter.link_availability = 100;
+                link_metrics.transmitter.phy_rate_mbps     = phy_rate_mbps;
+
+                link_metrics.receiver.packet_errors    = stats->rx_errors;
+                link_metrics.receiver.packets_received = stats->rx_packets;
+                link_metrics.receiver.rssi             = UINT8_MAX;
```

Comment:
This should be the default in the struct so not needed to set here (nitpick)

Reply:
RSSI field must be set to UINT8_MAX in Ethernet metrics only. It's not the default value